.github/workflows/maven.yml

```diff
@@ -33,4 +33,18 @@ jobs:
       run: mvn -B package -DskipTests --file pom.xml
     - name: Test with Maven
       run: mvn -B verify --file pom.xml
-    - uses: codecov/codecov-action@v2
+    - name: Add coverage to PR
```

Comment:
Is this name right? It says that it adds the code coverage to the PR, but this workflow happens not only in PRs, but also on push to the `main` branch (I believe that means that it also runs after it has been merged). What happens in that case?

Reply:
Hmmm... I'm not sure the documentation gives any explanation what would happen if you ran this on a push rather than a pull request. Might have to do some testing?
Either way, we need to run the tests in order to have a report to publish. Given we're currently running tests on main pushes and on PRs, if we don't want the report on pushes we'd need to split the workflow into two, one for each task, I would think?

Reply:
Yeah, as you said, we're not supposed to push to `main` directly, but by having the setting shown below, what I think it happens is that this workflow is triggered when something is added to main (not necessarily pushing directly, but as a result of merging a PR, for instance).
```yml
on:
  push:
    branches: [main]
```

Since you've tested this and it doesn't break anything, let's leave it as it is. We can always release a quick fix if we notice something strange.